### DISCUSSION TITLE
Overhead of combined tracing and logging

### DISCUSSION THREAD

**Discussion by rohel01** [bevyengine/bevy#10082] [Q&A]
Hello,

I am currently benchmarking my application using Bevy 0.11.2 with its `tracing-tracy` feature. My setup creates significant overhead, and I would welcome advice to optimize it.

First, the `tracing-tracy` feature seems to build on and reconfigure the logging infrastructure from the LogPlugin. So, when tracing is activated, every logged message now contains a (quite) long header. For example,

```log
1970-01-01T00:55:41.781836Z  INFO bevy_app:update:main app:schedule{name=Application}:system{name="svc::app::Application::run_application"}:schedule{name=Last}:multithreaded executor:system{name="svc::app::route_message"}: svc::app: 
```

The logging plugin attempts to write these long messages on `stdout`, so I have multiple systems trying to write long messages in parallel on the console, hence the overhead. I considered the following tweaks:

1. Deactivate logging to `stdout`: Did not find how to do that in Bevy documentation (incl. the -awesome- cheatbook) 
2. Change the format of the header so it is much shorter: Failed to find how to do that from the documentation
3. Raise the `level` setting of the `LogPlugin`so it just generates less logs: Works but Bevy built-in integration relies on INFO messages so I lose the automatic tracing ;( I still have my manually added spans though

I am running out of options, besides temporarily forking the relevant bevy crates. 

Advice welcomed!

**Comment by hymm** [comment]
if you don't need the console messages, you can get rid of most of the overhead by piping the output to a null file.
